kge/model/transd.py

```diff
@@ -0,0 +1,107 @@
+import torch
+from kge import Config, Dataset
+from kge.model.kge_model import RelationalScorer, KgeModel
+from torch.nn import functional as F
+from torch import Tensor
+from typing import List
+
+
+class TransDScorer(RelationalScorer):
+    r"""Implementation of the TransD KGE scorer."""
+
+    def __init__(self, config: Config, dataset: Dataset, configuration_key=None):
+        super().__init__(config, dataset, configuration_key)
+        self._norm = self.get_option("l_norm")
+
+    @staticmethod
+    def _transfer(ent_emb, ent_norm_vec_emb, rel_norm_vec_emb):
+        d = ent_norm_vec_emb.size(1)
+        k = rel_norm_vec_emb.size(1)
+
+        min_dim = min(d, k)
+
+        out = rel_norm_vec_emb * torch.sum(ent_norm_vec_emb * ent_emb, dim=-1, keepdim=True)
+        out[:, :min_dim] += ent_emb[:, :min_dim]
```

Comment:
Yes, I checked whether these two formula produce the same results. and honestly I referred to pykeen and other libraries. My original method was too slow because it used matrix calculations. So I changed this efficiently not to use matrix calculation.